crates/libcgroups/src/systemd/dbus_native/dbus.rs

```diff
@@ -0,0 +1,555 @@
+use crate::systemd::dbus_native::serialize::{DbusSerialize, Structure, Variant};
+
+use super::client::SystemdClient;
+use super::message::*;
+use super::proxy::Proxy;
+use super::utils::{DbusError, Result, SystemdClientError};
+use nix::sys::socket;
+use std::collections::HashMap;
+use std::io::{IoSlice, IoSliceMut};
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+const REPLY_BUF_SIZE: usize = 128; // seems good enough tradeoff between extra size and repeated calls
+
+/// NOTE that this is meant for a single-threaded use, and concurrent
+/// usage can cause errors, primarily because then the message received over
+/// socket can be out of order and we need to manager buffer and check with message counter
+/// which message is for which request etc etc
+// Client is a wrapper providing higher level API and abatraction around dbus.
+// For more information see https://www.freedesktop.org/wiki/Software/systemd/dbus/
+pub struct DbusConnection {
+    /// Is the socket system level or session specific
+    system: bool,
+    /// socket fd
+    socket: i32,
+    /// name id assigned by dbus for the connection
+    id: Option<String>,
```

Comment:
it is not necessary to send ID, but I have added it as sending it is a better practice